observability_product_sniffer.rb

```diff
@@ -19,39 +19,101 @@ def sniff
       #navigate to homepage
       driver.get(site)
 
+      ###################################################
+      ###################################################
+      ####### CHECK MAIN HOMEPAGE #######################
+      #####  FOR OBSERVABILITY TOOLS ####################
+      ###################################################
+      ###################################################
+
       if !tools_used.empty?
-        detected_on = 'homepage'
-      else
-        # if no observability tool was detected on the homepage,
-        # navigate to login/account/sign in page, because these are more
-        # likely to contain app code and therefore observability monitoring.
-
-        # using elements plural here to avoid error-throwing, and then jsut selecting first result
-        sign_in = driver.find_elements(:xpath, "//a[contains(translate(.,'ABCDEFGHIJKLMNOPQRSTUVWXYZ','abcdefghijklmnopqrstuvwxyz'),'sign in')]").map{|el| el.attribute('href')}
-        log_in = driver.find_elements(:xpath, "//a[contains(translate(.,'ABCDEFGHIJKLMNOPQRSTUVWXYZ','abcdefghijklmnopqrstuvwxyz'),'log in')]").map{|el| el.attribute('href')}
-        login = driver.find_elements(:xpath, "//a[contains(translate(.,'ABCDEFGHIJKLMNOPQRSTUVWXYZ','abcdefghijklmnopqrstuvwxyz'),'login')]").map{|el| el.attribute('href')}
-        account = driver.find_elements(:xpath, "//a[contains(translate(.,'ABCDEFGHIJKLMNOPQRSTUVWXYZ','abcdefghijklmnopqrstuvwxyz'),'account')]").map{|el| el.attribute('href')}
-
-        url = nil
-        if !sign_in.empty?
-          detected_on = 'sign in'
-          url = sign_in.first
-        elsif !log_in.empty?
-          detected_on = 'log in'
-          url = log_in.first
-        elsif !login.empty?
-          detected_on = 'login (no space)'
-          url = login.first
-        elsif !account.empty?
-          detected_on = 'account'
-          url = account.first
-        else
-          detected_on = 'n/a'
+        return {
+          tools_used: tools_used,
+          detected_on: tools_used.empty? ? "n/a" : 'homepage',
+          error: nil
+        }
+      end
+
+      ###################################################
+      ###################################################
+      ###### CHECK ALL HOMEPAGE FRAMES ##################
+      ######  FOR OBSERVABILITY TOOLS ###################
+      ###################################################
+      ###################################################
+
```

Comment:
On L43, add a sleep(5) or 10seconds so we allow the iframes to load before detecting

Give a few instances / print out the URL where this happens so Neil can look at the DOM. Print out homepage URL where this errors.